[v14] Fix the RDP licensing flow

zmb3 opened the pull request:
The actual fixes were implemented in rdp-rs, this PR just updates the dependency and API interface to accomodate the new library version.

- https://github.com/gravitational/rdp-rs/pull/42
- https://github.com/gravitational/rdp-rs/pull/43

Changelog: Fixed an issue with the Microsoft license negotiation for RDP sessions.

zmb3 committed:
Fix the RDP licensing flow on v14

Licenses are stored in-memory, so if the agent restarts it will be
forced to go through the "new license" flow for the first session.

probakowski commented on lib/srv/desktop/rdp/rdpclient/src/lib.rs:
```diff
@@ -2048,6 +2063,57 @@ pub struct CGOSharedDirectoryListRequest {
+        }
+    }
+
+    #[allow(dead_code)]
+    pub fn into_inner(self) -> L {
+        self.inner.into_inner().unwrap()
+    }
```
Do we need this?

espadolini replied:
It might be dead code because the type itself is not public; being internal-use only I guess we can remove it and add it if and when we end up needing it?

zmb3 commented on lib/srv/desktop/rdp/rdpclient/client.go:
```diff
@@ -237,12 +237,14 @@ func (c *Client) connect(ctx context.Context) error {
 	}
 
-	// Addr and username strings only need to be valid for the duration of
+	// These strings only need to be valid for the duration of
 	// C.connect_rdp. They are copied on the Rust side and can be freed here.
```
I suppose we could pin them instead without copying into a C allocation, but I don't feel the need to change this now given where v14 is in its lifecycle.

zmb3 replied:
Ah, I suppose we're making the C allocation so that we can pass a null terminated string.

Doing this without allocation would require we change the Cgo call to pass both a pointer and a length.

zmb3 committed:
Store RDP licenses in a LRU cache instead of a HashMap

This ensures memory doesn't grow unbounded if we end up caching
a large number of licenses.
